generator/src/main/java/org/corfudb/generator/Generator.java

```diff
@@ -19,55 +19,201 @@
         Runnable cpTrimTask = () -> {
             Operation op = new CheckpointOperation(state);
             op.execute();
         };
-
-        final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
         scheduler.scheduleAtFixedRate(cpTrimTask, 30, cpPeriod, TimeUnit.SECONDS);
+    }
 
-        ExecutorService appWorkers = newWorkStealingPool(numThreads);
 
-        Runnable app = () -> {
-            List<Operation> operations = state.getOperations().sample(numOperations);
-            for (Operation operation : operations) {
-                try {
-                    operation.execute();
-                } catch (Exception ex) {
-                    throw new RuntimeException(ex);
+    public void shutdown() {
+        workers.shutdown();
+        scheduler.shutdown();
+    }
+
+
+    /**
+     * Run a Longevity test that 'freezes' the system every CHUNK_RUNTIME,
+     * sends the correctness log (of one chunk) to verification, then resumes
+     * the run.
+     * The same action happens in a cyclic fashion until the totalRuntime.
+     */
+    public void runLongevityTestForVerification() {
+
+        fillOperationQueue();
+        long lastPollTime = System.currentTimeMillis();
+        consumeOperationQueue();
+        runCpTrimOperation();
+
+        while (true) {
+            // One round of 'chunk' starts
+            long cumulativeRuntime = 0;
+            while (true) {
+                Sleep.sleepUninterruptibly(POLL_TIME);
+
+                // Check if the operationQueue is drained or not,
+                // if it's empty and has not reached CHUNK_RUNTIME,
+                // then fill up the queue for further consuming.
+                if (operationQueue.isEmpty()) {
+                    // Calculate the 'cumulative' runtime so far,
+                    // skipping the time for filling up the operation queue
+                    cumulativeRuntime += System.currentTimeMillis() - lastPollTime;
+                    // Break this while loop if it reaches CHUNK_RUNTIME
+                    if (cumulativeRuntime >= CHUNK_RUNTIME.toMillis()) {
+                        // At the end of the chunk, give some margin
+                        // for the workers to finish their operations
+                        Sleep.sleepUninterruptibly(CHUNK_TIMEOUT);
+                        Correctness.recordOperation("Liveness, " + "Success", false);
+                        break;
+                    }
+                    fillOperationQueue();
+                    lastPollTime = System.currentTimeMillis();
+                } else {
+                    // If it's been more than LIVENESS_TIMEOUT, and the queue is still not drained,
+                    // then the system might have got stuck and we record its 'Liveness' as 'Fail'.
+                    if (System.currentTimeMillis() - lastPollTime > LIVENESS_TIMEOUT.toMillis()) {
+                        Correctness.recordOperation("Liveness, " + "Fail", false);
```

Comment:
We need to stop these consumers that are failed, otherwise the failed consumers are still running and cannot be scheduled to do the operations in the next chunk.

Reply:
As I explained before, we should let the human to decide whether to stop a test or not. And here may not be the consumers' fault, i.e. the system itself got stuck can lead to the problem. We will have a correctness report reflecting the 'liveness failure', which gives enough indication for the person who run the test.